File: mycollab-dao/src/main/scala/com/mycollab/db/arguments/NumberSearchField.scala
package com.mycollab.db.arguments

import scala.beans.BeanProperty

/**
  * @author MyCollab Ltd
  * @since 5.3.5
  */
class NumberSearchField(operation: String, @BeanProperty val value: Number,
                        @BeanProperty val compareOperator: String) extends SearchField(operation) {
  def this(value: Number) = this(SearchField.AND, value, NumberSearchField.EQUAL)
  
  def this(value: Number, compareOperator: String) = this(SearchField.AND, value, compareOperator)
}

object NumberSearchField {
  val EQUAL = "="
  val NOT_EQUAL = "<>"
  val LESS_THAN = "<"
  val GREATER = ">"
  
  def equal(value: Number): NumberSearchField = new NumberSearchField(SearchField.AND, value, EQUAL)
  
  def greaterThan(value: Number): NumberSearchField = new NumberSearchField(SearchField.AND, value, GREATER)
  
  def lessThan(value: Number): NumberSearchField = new NumberSearchField(SearchField.AND, value, LESS_THAN)
}

File: mycollab-dao/src/main/java/com/mycollab/db/arguments/NumberSearchField.kt
package com.mycollab.db.arguments

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
open class NumberSearchField constructor(operation: String, var value: Int?, var compareOperator: String) : SearchField(operation) {
    constructor(value: Int?) : this(SearchField.AND, value, NumberSearchField.EQUAL)

    constructor(value: Int?, compareOperator: String) : this(SearchField.AND, value, compareOperator)

    companion object {
        @JvmField val EQUAL = "="
        @JvmField val NOT_EQUAL = "<>"
        @JvmField val LESS_THAN = "<"
        @JvmField val GREATER = ">"

        @JvmStatic fun equal(value: Int) = NumberSearchField(SearchField.AND, value, EQUAL)

        @JvmStatic fun greaterThan(value: Int) = NumberSearchField(SearchField.AND, value, GREATER)

        @JvmStatic fun lessThan(value: Int) = NumberSearchField(SearchField.AND, value, LESS_THAN)
    }
}